codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexHandler.java

```diff
@@ -123,12 +131,47 @@ public Http2MultiplexHandler(ChannelHandler inboundStreamHandler) {
      *                             the {@link Channel}s created for new inbound streams.
      * @param upgradeStreamHandler the {@link ChannelHandler} that will be added to the {@link ChannelPipeline} of the
      *                             upgraded {@link Channel}.
+     * @deprecated Use {@link #forClient(ChannelHandler, ChannelHandler)} instead.
      */
+    @Deprecated
```

Comment:
is deprecation actually necessary? This change helps for small edge case (testing, with EmbeddedChannel only - but many use real channels for testing), otherwise current API is sufficient. It would force netty users onto new API for no practical gain.